abaci.md

```diff
@@ -0,0 +1,158 @@
+            <abacus-wards>   SetItem(MSGSENDER) </abacus-wards>
+            ...
+           </abacus>
+         )
+         [owise]
+```
+
+Abacus Authorization
+------------------
+
+```k
+    syntax AbacusStep ::= AbacusAuthStep
+    syntax AuthStep   ::= AbacusContract "." AbacusAuthStep [klabel(abacusStep)]
+ // ----------------------------------------------------------------------------
+    rule [[ wards(Abacus ABACUS_ADDRESS) => WARDS ]] <abaci> <abacus-address> ABACUS_ADDRESS </abacus-address> <abacus-wards> WARDS </abacus-wards> ... </abaci>
+
+    syntax AbacusAuthStep ::= WardStep
+ // --------------------------------
+    rule <k> Abacus ABACUS_ADDRESS . rely ADDR => . ... </k>
+         <abacus>
+           <abacus-address>   ABACUS_ADDRESS              </abacus-address>
+           <abacus-wards>     ... (.Set => SetItem(ADDR)) </abacus-wards>
+           ...
+         </abacus>
+
+    rule <k> Abacus ABACUS_ADDRESS . deny ADDR => . ... </k>
+         <abacus>
+           <abacus-address>   ABACUS_ADDRESS                    </abacus-address>
+           <abacus-wards>     WARDS => WARDS -Set SetItem(ADDR) </abacus-wards>
+           ...
+         </abacus>
+```
+
+
+File-able Fields
+----------------
+
+These parameters are controlled by governance:
+
+-   `tau`:  Seconds after auction start when the price reaches zeroe
+-   `step`: Length of time between price drops
+-   `cut`:  Per-step multiplicative facto
+
+
+```k
+    syntax AbacusAuthStep ::= "file" AbacusFile
+ // --------------------------------------------
+
+    syntax AbacusFile ::= "tau"  Int
+                        | "step" Int
+                        | "cut"  Ray
+ // --------------------------------------------
+    rule <k> Abacus ABACUS_ADDRESS . file tau TAU => . ... </k>
+         <abacus>
+            <abacus-address> ABACUS_ADDRESS </abacus-address>
+            <abacus-tau>     _ => TAU       </abacus-tau>
+            ...
+         </abacus>
+
+    rule <k> Abacus ABACUS_ADDRESS . file step STEP => . ... </k>
+         <abacus>
+            <abacus-address> ABACUS_ADDRESS </abacus-address>
+            <abacus-step>    _ => STEP      </abacus-step>
+            ...
+         </abacus>
+
+    rule <k> Abacus ABACUS_ADDRESS . file cut CUT => . ... </k>
+         <abacus>
+            <abacus-address> ABACUS_ADDRESS </abacus-address>
+            <abacus-cut>     _ => CUT       </abacus-cut>
+            ...
+         </abacus>
+```
+
+
+Abacus Semantics
+----------------
+
+```k
+   syntax AbacusStep ::= "price" Ray Int
+ // -------------------------------------
+   rule <k> Abacus ABACUS_ADDRESS . price TOP DUR => ( TOP *Ray ray( ( ABACUS_TAU -Int DUR ) /Int ABACUS_TAU ) ) ... </k>
+      <abacus>
+         <abacus-address> ABACUS_ADDRESS   </abacus-address>
+         <abacus-type>    "LinearDecrease" </abacus-type>
+         <abacus-tau>     ABACUS_TAU       </abacus-tau>
+         ...
+      </abacus>
+      requires DUR >=Int ABACUS_TAU
```

Comment:
I think this condition should actually be `<=Int`. When `DUR > ABACUS_TAU`, the correct behavior is to evaluate to zero. Of course we don't use the linear decrease in practice anyway, so you could arguably omit it.

Reply:
Checked on abaci.sol, fixed the condition and added a second rule for when it evaluates to 0